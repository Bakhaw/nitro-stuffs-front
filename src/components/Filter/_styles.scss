.Filter {
  margin-bottom: 25px;
  opacity: 0.3;
  cursor: pointer;
  transition: 0.8s cubic-bezier(0.2, 0.8, 0.2, 1);

  &:hover {
    opacity: 1;
    transition: 0.8s cubic-bezier(0.2, 0.8, 0.2, 1);
  }
}

.Filter-active {
  opacity: 1;
}
