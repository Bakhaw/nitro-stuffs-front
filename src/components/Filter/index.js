import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import { StateContext } from '../../Context';

function Filter({ active = false, filter, handleFilterClick }) {
  const { currentGame } = useContext(StateContext);

  const backgroundURL = require(`../../static/images/${currentGame}/stuff-icons/${filter.type}.svg`);
  const backgroundImage = `url(${backgroundURL})`;

  return (
    <div
      className={`Filter ${active ? 'Filter-active' : ''}`}
      onClick={() => handleFilterClick(filter)}
      style={{ backgroundImage }}
    />
  );
}

Filter.propTypes = {
  active: PropTypes.bool,
  filter: PropTypes.object.isRequired,
  handleFilterClick: PropTypes.func.isRequired,
};

export default Filter;
