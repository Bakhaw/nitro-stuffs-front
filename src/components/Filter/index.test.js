import React from 'react';
import { shallow } from 'enzyme';

import Filter from '.';

test('should render properly', () => {
  const container = shallow(<Filter />);
  expect(container.debug()).toMatchSnapshot();
});
