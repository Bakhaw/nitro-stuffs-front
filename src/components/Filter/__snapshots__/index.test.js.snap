// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
"<ContextProvider value={{...}}>
  <Filter active={false} filter={{...}} handleFilterClick={[Function: mockConstructor]} />
</ContextProvider>"
`;
